sweepai/core/code_repair.py

```diff
@@ -47,7 +47,12 @@ def check_syntax(old_code, file_extension: str) -> bool:
     def repair_code(self, diff: str, user_code: str, feature:str) -> str:
         self.messages = [Message(role="system", content=code_repair_system_prompt.format(feature=feature))]
         self.model = "gpt-3.5-turbo-16k-0613" # can be optimized
-        response = self.chat(code_repair_prompt.format(diff=diff, user_code=user_code))
+        while True:
+            response = self.chat(code_repair_prompt.format(diff=diff, user_code=user_code))
+            # Check if the length of the response does not differ by more than 10% from the input
+            if len(user_code.splitlines()) > 50 and abs(len(response.splitlines()) - len(user_code.splitlines())) / len(user_code.splitlines()) > 0.1:
+                continue
```

Comment:
add the undo if we retry, also cause this to terminate after 5 tries instead of a while loop. if it terminates, return the user_code directly